tiledb/api/c_api/domain/domain_api.cc

```diff
@@ -142,12 +144,39 @@ int32_t tiledb_domain_has_dimension(
 
 int32_t tiledb_domain_dump(const tiledb_domain_t* domain, FILE* out) {
   ensure_domain_is_valid(domain);
-  domain->dump(out);
+
+  if (out == nullptr)
+    return TILEDB_ERR;
+
+  std::stringstream ss;
+  ss << *domain;
+  size_t r = fwrite(ss.str().c_str(), sizeof(char), ss.str().size(), out);
+  if (r != ss.str().size()) {
+    throw CAPIException("Error writing domain to output stream");
+  }
+
+  return TILEDB_OK;
+}
+
+int32_t tiledb_domain_dump_str(
+    const tiledb_domain_t* domain, tiledb_string_t** out) {
```

Comment:
```suggestion
    const tiledb_domain_t* domain, tiledb_string_handle_t** out) {
```

Reply:
Couldn't achieve this. Should be `struct tiledb_string_handle_t` if you want to use this here and I suppose that typedef is used for exactly these cases.

As I can see handles are only used in the files that they are defined. Every other file uses their typedef.